Author: Pino Toscano
Description: fixed checks for hurd in configure and Makefile.am
 TODO: while autoconf checks for pthread and dl, it never uses the results...
 for now we hardcode them
Forward: https://sourceforge.net/tracker/?func=detail&aid=3297657&group_id=55736&atid=478072
--- puredata.orig/configure.ac
+++ puredata/configure.ac
@@ -37,11 +37,8 @@
 	LDFLAGS="$LDFLAGS -L/sw/lib"
 	EXTERNAL_LDFLAGS="-bundle -undefined dynamic_lookup"
 	;;
-*linux*|GNU|*kfreebsd*gnu*)
-# GNU and GNU/kFreeBSD are for Debian, were they are treated very similar to linux
-	if test "x${host}" = "xGNU"; then
-	 HURD=yes
-	fi
+*linux*|*kfreebsd*gnu*)
+# GNU/kFreeBSD are for Debian, were they are treated very similar to linux
 	if test "x${ANDROID}" = "xno"; then
 	 LINUX=yes
 	 portaudio=yes
@@ -50,6 +47,13 @@
 	EXTERNAL_CFLAGS="-fPIC"
 	EXTERNAL_LDFLAGS="-Wl,--export-dynamic -shared -fPIC"
 	EXTERNAL_EXTENSION=pd_linux
+	;;
+*-*-gnu*)
+	HURD=yes
+	CFLAGS="-O6 -funroll-loops -fomit-frame-pointer $CFLAGS"
+	EXTERNAL_CFLAGS="-fPIC"
+	EXTERNAL_LDFLAGS="-Wl,--export-dynamic -shared -fPIC"
+	EXTERNAL_EXTENSION=pd_linux
 	;;
 *mingw*)
 	WINDOWS=yes
--- puredata.orig/src/Makefile.am
+++ puredata/src/Makefile.am
@@ -106,7 +106,18 @@
 endif
 
 if HURD
-pd_SOURCES += s_midi_dummy.c
+libpdbindir = $(pkglibdir)/bin
+libpdbin_DATA =
+libpdbin_PROGRAMS = pd-watchdog pd
+# this flag has to have a single leading "-" for libtool, even though ld uses
+# --export-dynamic, and libtool sends -Wl,--export-dynamic to ld...
+pd_LDFLAGS += -export-dynamic
+# on Ubuntu/Karmic 9.10, it doesn't seem to find libm, so force it
+pd_LDFLAGS += $(LIBM)
+# force linking to pthread, which should really be done with some autotools way
+pd_LDFLAGS += -lpthread
+# force linking to dl, which should really be done with some autotools way
+pd_LDFLAGS += -ldl
 endif
 
 if LINUX 
